popsycle/synthetic.py

```diff
@@ -3512,3 +3527,394 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
+            p = np.outer(p,pnfwunorm(1, con=con))
+        else:
+            p *= pnfwunorm(1, con=con)
+        return (-(1.0/np.real(special.lambertw(-np.exp(-p-1))))-1)/con
+
+    #Calculating radius values for all PBHs within 8.3kpc of the galactic center
+    r_values = (qnfw(np.random.rand(int(num_pbh_8kpc)) * pnfw(r_max/r_vir,con=c, logp=False), con=c)*r_vir)
+
+    #Sample PBH lattitude and longitude to have full spherical coordinates.
+    sin_lats = np.random.uniform(-1, 1, int(num_pbh_8kpc))
+    lats=np.arcsin(sin_lats)
+    longs = np.random.uniform(0, 2*np.pi, int(num_pbh_8kpc))
+
+    #Converting spherical galactocentric coordinates to cartesian galactocentric coordinates
+    cart = astropy.coordinates.spherical_to_cartesian(r_values, lats, longs)
+
+    #Defining galactocentric coordinates
+    c = coord.Galactocentric(x=cart[0] * units.kpc, y=cart[1] * units.kpc, z=cart[2] * units.kpc)
+
+    #Transforming from galactocentric to galactic coordinates
+    #outputs l, b, dist in degrees.
+    c = c.transform_to(coord.Galactic(representation_type='cartesian'))
+
+    #Set minimum and maximum l, b, and r values for PBH mask. 
+    #Determining the center of the field of view circle.
+    l_min = np.min(long_bin).values
+    l_max = np.max(long_bin).values
+    b_min = np.min(lat_bin).values
+    b_max = np.max(lat_bin).values
+    l_mid = np.average((l_min, l_max))
+    b_mid = np.average((b_min, b_max))
+
+    dists = angdist(l_mid, b_mid, c.l.deg, c.b.deg)
+    r_max_mask = 16.6
+
+    #Masking the full PBH data for our field of view
+    mask = (c.l.deg >= l_min) & (c.l.deg <= l_max) & (c.b.deg >= b_min) & (c.b.deg <= b_max) & (c.distance.kpc <= r_max_mask) & (dists < field_of_view_area)
+    data_in_field = c[mask]
+
+    #Obtain the radius values for all of the PBHs in our field of view.
+    r = data_in_field.distance.kpc
+    l = data_in_field.l.deg
+    b = data_in_field.b.deg
+
+    c_pbh = coord.Galactic(l=l * u.deg, b=b * u.deg, distance=r * u.kpc)
+    c_pbh = c_pbh.transform_to(coord.Galactocentric(representation_type='spherical'))
+    cart_pbh = astropy.coordinates.cartesian_to_spherical(c_pbh.x, c_pbh.y, c_pbh.z)
+    pbh_r_galacto = cart[0]
+
+    if inner_slope == 1:
+        vel_data = pd.read_csv('data/red_line.csv')
+    elif inner_slope == .25:
+        vel_data = pd.read_csv('data/blue_line.csv')
+    else:
+        vel_data = pd.read_csv('data/green_line.csv')
+
+    pbh_vrms = np.interp(pbh_r_galacto, vel_data['x'], vel_data['y'])
+    v_vals = np.arange(0, v_esc) #Goes from v to v_esc
+    a = (1/2)*pbh_vrms*((np.pi/2)**(1/2))
+
+    for a_val in a:
+        pdf = ((2/np.pi)**(1/2))*((v_vals**2*np.exp(-v_vals**2/(2*a_val**2)))/a_val**3)
+        cdf = scipy.special.erf(v_vals/(a_val*2**(1/2)))-(((2/np.pi)**(1/2))*((v_vals*np.exp(-v**2/2*a_val**2))/a_val))
+        rand_cdf = np.random.uniform(0, np.amax(cdf), 1)
+        interpreted_rms_velocities = np.interp(rand_cdf, cdf, v_vals)
```

Comment:
interpreted_rms_velocities should be an array that is filled up on each iteration of the loop

Reply:
Fixed. It will be reflected in next push.